src/Swarm/Language/Syntax.hs

```diff
@@ -1001,6 +1009,7 @@ freeVarsS f = go S.empty
     SDef r x xty s1 -> rewrap $ SDef r x xty <$> go (S.insert (lvVar x) bound) s1
     SBind mx s1 s2 -> rewrap $ SBind mx <$> go bound s1 <*> go (maybe id (S.insert . lvVar) mx bound) s2
     SDelay m s1 -> rewrap $ SDelay m <$> go bound s1
+    SAnnotate s1 _ -> go bound s1
```

Comment:
This should be something more like `rewrap $ SAnnotate <$> go bound s1 <*> pure ty` (untested).  There's no reason that traversing over free variables should remove `SAnnotate` nodes.

Reply:
I completely agree. Thank you for pointing out this mistake. I think that your proposed fix works and will change this line accordingly.